### ISSUE TITLE
VeraCrypt in docker container: Mount VeraCrypt to to docker volume mount - Error Mount point is already in use

### ISSUE THREAD

**Issue by tarchive** [veracrypt/VeraCrypt#464]
Hello, 

I am using docker image https://hub.docker.com/r/dcflachs/veracrypt-gui/ on my Synology NAS 

I require three different connections with R/W access at any given time to the VeraCrypt volume and due to limitation where multiple mounts can only be R/O, i must only mount this once and share the mounted directory to the other clients. 

The trouble i am having is that the folder i attach from the host to the docker container is not a valid mounting point for veracrypt. It will give me the error "mount point is already in use" message. The only way i can get the container to mount is by picking a non-docker folder.  I have tried bind mounts and symlinks but they do not work outside of the container.

Basically, on the Synolgy NAS host i have the following folders:
Synology CIFS share: /volume1/veracrypt
VeraCrypt volume file: /volume/veracrypt/in/encryptedfile
VeraCrypt mount folder: /volume/vercrypt/out

docker volume settings
-v /volume1/veracrypt/in:/mnt/in
-v /volume1/veracrypt/out:/mnt/out

VeraCrypt setting in docker container
Volume path: /mnt/in/encryptedfile
mount path:/mnt/out

Is there a trick to getting VeraCrypt to mount to docker host folders? Or should i be approaching this differently? Thanks.

**Comment by tianon** [comment]
To access the container mounts externally via bind mount, you likely need to mess with Docker's "bind propagation" functionality (https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation).

**Comment by dcflachs** [comment]
@tarchive Please forgive me for resurrecting an old topic, I hope you already got this to work. If you look closely at the docker hub page it is mentioned that the docker mount in which you which to mount VeraCrypt must have the **shared** propagation setting. I probably should adjust the docker hub readme to make it more obvious/

**Comment by tarchive** [comment]
Thank you both for the replies, i did not receive notification or i would have replied sooner. 

No unfortunately i stepped away from trying to use VeraCrypt on my Synology NAS in favor of whole disk encryption (LUKS) on an UNRAID NAS. Since my original comment i have removed all my test containers and image files. 
For the sake of closing this ticket, i have decided to give it one more attempt using your suggested shared propagation setting.
Unfortunately i hit another roadblock. The container won't start and gives the following error 
"Error response from daemon: linux mounts: path /volume1/VeraCrypt is mounted on /volume1 but it is not a shared mount"
There are a few articles written up describing a workaround but i do not see how to apply it to my Synology NAS.
https://success.docker.com/article/not-a-shared-mount-error
https://success.docker.com/article/install-ucp-3-on-old-distro

I'll poke around some more but just wanted to give an update for now.
